### DISCUSSION TITLE
is there a way to change nlist parameter of an existing collection?

### DISCUSSION THREAD

**Discussion by buntys2010** [milvus-io/milvus#22805] [Q&A and General discussion] [answered]
I have a collection created with 1024 nlist parameter but now i want to change the nlist parameter to 4096 of an existing collection.

Could someone please advise if there is a way or the only way is to delete the collection and re-create it again?

**Comment by xiaofan-luan** [comment, marked as answer]
release collection
drop index
recreate index

**Reply by buntys2010** [reply]
Thanks for your quick response, could you please guide me to a documentation how can i perform this activity via Python API?

**Reply by xiaofan-luan** [reply]
from pymilvus import Collection
collection = Collection("book")      # Get an existing collection.
collection.release()
collection.drop_index()
collection.create_index(
  field_name="book_intro", 
  index_params=index_params
)
